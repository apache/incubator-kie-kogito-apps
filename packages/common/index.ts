export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as PageLayout
} from './src/components/Templates/PageLayout/PageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as BrandLogo
} from './src/components/Atoms/BrandLogo/BrandLogo';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export {
  default as KogitoEmptyState
} from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
