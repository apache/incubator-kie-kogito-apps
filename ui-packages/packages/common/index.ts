export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
