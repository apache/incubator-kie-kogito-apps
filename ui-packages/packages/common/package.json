{
  "name": "@kogito-apps/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "scripts": {
    "precommit": "lint-staged",
    "build:prod": "yarn run lint && yarn run test:coverage",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "update-snapshot": "jest --updateSnapshot",
    "test:coverage": "yarn test --coverage",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist",
    "codegen": "graphql-codegen"
  },
  "dependencies": {},
  "devDependencies": {
    "@storybook/addon-a11y": "5.3.19",
    "@storybook/addon-actions": "5.3.18",
    "@storybook/addon-backgrounds": "5.3.19",
    "@storybook/addon-console": "1.2.1",
    "@storybook/addon-knobs": "5.3.18",
    "@storybook/addon-links": "5.3.18",
    "@storybook/addon-options": "5.3.18",
    "@storybook/addon-storysource": "5.3.18",
    "@storybook/addon-viewport": "5.3.18",
    "@storybook/react": "5.3.18",
    "@storybook/source-loader": "5.3.18",
    "babel-loader": "8.1.0",
    "react-docgen-typescript-loader": "3.7.2",
    "storybook-react-router": "1.0.8"
  },
  "jest": {
    "preset": "ts-jest/presets/js-with-ts",
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "coveragePathIgnorePatterns": [
      "./src/static",
      "./src/graphql"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}
