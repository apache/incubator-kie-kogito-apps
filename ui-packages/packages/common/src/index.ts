/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export {
  default as DataTable,
  DataTableColumn
} from './components/Organisms/DataTable/DataTable';
export { default as KogitoSpinner } from './components/Atoms/KogitoSpinner/KogitoSpinner';
export { default as PageToolbar } from './components/Molecules/PageToolbar/PageToolbar';
export { default as KogitoPageLayout } from './components/Templates/KogitoPageLayout/KogitoPageLayout';
export { default as AboutModalBox } from './components/Molecules/AboutModalBox/AboutModalBox';
export { default as EndpointLink } from './components/Atoms/EndpointLink/EndpointLink';
export { default as PageNotFound } from './components/Molecules/PageNotFound/PageNotFound';
export { default as ServerUnavailable } from './components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './components/Molecules/NoData/NoData';
export { default as ServerErrors } from './components/Molecules/ServerErrors/ServerErrors';
export { default as ItemDescriptor } from './components/Molecules/ItemDescriptor/ItemDescriptor';
export * from './components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './components/Atoms/LoadMore/LoadMore';
export { default as DomainExplorer } from './components/Organisms/DomainExplorer/DomainExplorer';
export { default as DomainExplorerListDomains } from './components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './utils/Utils';
export * from './environment/auth/Auth';
export {
  default as KogitoAppContext,
  AppContext,
  useKogitoAppContext
} from './environment/context/KogitoAppContext';
export * from './utils/KeycloakClient';
export * from './graphql/types';
export { default as KogitoAppContextProvider } from './components/Molecules/KogitoAppContextProvider/KogitoAppContextProvider';
