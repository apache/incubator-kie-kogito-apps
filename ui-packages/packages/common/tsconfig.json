{
    "extends": "../../tsconfig.json",
    "compilerOptions": {
        "declarationDir": "./dist",
        "declaration": true
    },
    "files": [
        "./index.ts"
    ],
    "include": [
        "src/",
        "../../config/declare.d.ts"
    ],
    "exclude": [
        "src/graphql/queries.tsx",
        "src/**/*.test.ts",
        "src/**/*.test.tsx",
        "src/**/*.stories.tsx",
        "src/**/__mocks__"
    ] 
}
