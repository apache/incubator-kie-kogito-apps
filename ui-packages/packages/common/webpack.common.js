const path = require('path');
const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
const BG_IMAGES_DIRNAME = 'bgimages';

module.exports = {
  entry: {
    index: path.resolve(__dirname, 'index.ts')
  },
  devtool: "inline-source-map",
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].bundle.js',
    libraryTarget: "umd",
    globalObject: "this"
  },
  module: {
    rules: [
      {
        test: /\.(tsx|ts)?$/,
        use: [
          {
            loader: 'ts-loader',
            options: {
              configFile: path.resolve('./tsconfig.json'),
            }
          }
        ]
      },
      {
        test: /\.(svg|jpg|ttf|eot|woff|woff2)$/,
        include: [
          path.resolve('../../node_modules/patternfly/dist/fonts'),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/fonts'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/pficon'
          ),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/fonts'
          ),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/pficon'
          ),
          path.resolve('./src/static')
        ],
        use: {
          loader: 'file-loader',
          options: {
            // Limit at 50k. larger files emited into separate files
            limit: 5000,
            outputPath: 'src/static',
            name: '[name].[ext]'
          }
        }
      },
      {
        test: /\.svg$/,
        include: input => input.indexOf('background-filter.svg') > 1,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'svgs',
              name: '[name].[ext]'
            }
          }
        ]
      },
      {
        test: /\.svg$/,
        include: input => input.indexOf(BG_IMAGES_DIRNAME) > -1,
        use: {
          loader: 'svg-url-loader',
          options: {}
        }
      },
      {
        test: /\.(jpg|jpeg|png|gif)$/i,
        include: [
          path.resolve(__dirname, 'src'),
          path.resolve('../../node_modules/patternfly'),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve('../../../../../../packages/common/src/static/'),
          path.resolve('./src/static'),
        ],
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'src/static',
              name: '[name].[ext]'
            }
          }
        ]
      }
    ]
  },
  resolve: {
    extensions: ['.ts', '.tsx', '.js'],
    modules: [
      path.resolve('../../node_modules'),
      path.resolve('./node_modules'),
      path.resolve('./src')
    ],
    plugins: [
      new TsconfigPathsPlugin({
        configFile: path.resolve(__dirname, './tsconfig.json')
      })
    ],
    symlinks: false,
    cacheWithContext: false
  }
};