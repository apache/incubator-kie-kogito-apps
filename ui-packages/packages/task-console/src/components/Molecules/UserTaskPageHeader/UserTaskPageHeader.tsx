import React from 'react';
import { Link } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbItem,
  PageSection
} from '@patternfly/react-core';

import PageTitle from '../PageTitle/PageTitle';

const UserTaskPageHeader: React.FC = () => {
  return (
    <React.Fragment>
      <PageSection variant="light">
        <PageTitle title="User Tasks" />
        <Breadcrumb>
          <BreadcrumbItem>
            <Link to={'/'}>Home</Link>
          </BreadcrumbItem>
          <BreadcrumbItem isActive>User Tasks</BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
    </React.Fragment>
  );
};

export default UserTaskPageHeader;
