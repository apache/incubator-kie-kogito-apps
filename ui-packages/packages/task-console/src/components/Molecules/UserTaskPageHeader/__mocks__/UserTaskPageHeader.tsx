import React from 'react';

const MockUserTaskPageHeader = (): React.ReactElement => {
  return <></>;
};

export default MockUserTaskPageHeader;
