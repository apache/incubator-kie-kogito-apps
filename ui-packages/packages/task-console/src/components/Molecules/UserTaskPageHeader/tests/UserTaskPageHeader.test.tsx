import React from 'react';
import UserTaskPageHeader from '../UserTaskPageHeader';
import { getWrapper } from '@kogito-apps/common';
import { BrowserRouter } from 'react-router-dom';

describe('UserTaskPageHeader component tests', () => {
  it('Should render UserTaskPageHeader correctly', () => {
    const wrapper = getWrapper(
      <BrowserRouter>
        <UserTaskPageHeader />
      </BrowserRouter>,
      'UserTaskPageHeader'
    );
    expect(wrapper).toMatchSnapshot();
  });
});
