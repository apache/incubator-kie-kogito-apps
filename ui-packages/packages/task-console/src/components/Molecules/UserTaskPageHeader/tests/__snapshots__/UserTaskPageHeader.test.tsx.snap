// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTaskPageHeader component tests Should render UserTaskPageHeader correctly 1`] = `
<UserTaskPageHeader>
  <PageSection
    variant="light"
  >
    <section
      className="pf-c-page__main-section pf-m-light"
    >
      <PageTitle
        title="User Tasks"
      >
        <Title
          headingLevel="h1"
          size="4xl"
        >
          <h1
            className="pf-c-title pf-m-4xl"
          >
            User Tasks
          </h1>
        </Title>
      </PageTitle>
      <Component>
        <ComponentWithOuia
          component={[Function]}
          componentProps={
            Object {
              "children": Array [
                <BreadcrumbItem>
                  <Link
                    to="/"
                  >
                    Home
                  </Link>
                </BreadcrumbItem>,
                <BreadcrumbItem
                  isActive={true}
                >
                  User Tasks
                </BreadcrumbItem>,
              ],
            }
          }
          consumerContext={null}
        >
          <Breadcrumb
            ouiaContext={
              Object {
                "isOuia": false,
                "ouiaId": null,
              }
            }
          >
            <nav
              aria-label="Breadcrumb"
              className="pf-c-breadcrumb"
            >
              <ol
                className="pf-c-breadcrumb__list"
              >
                <BreadcrumbItem>
                  <li
                    className="pf-c-breadcrumb__item"
                  >
                    <Link
                      to="/"
                    >
                      <LinkAnchor
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                      </LinkAnchor>
                    </Link>
                    <span
                      className="pf-c-breadcrumb__item-divider"
                    >
                      <AngleRightIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 256 512"
                          width="1em"
                        >
                          <path
                            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                            transform=""
                          />
                        </svg>
                      </AngleRightIcon>
                    </span>
                  </li>
                </BreadcrumbItem>
                <BreadcrumbItem
                  isActive={true}
                >
                  <li
                    className="pf-c-breadcrumb__item"
                  >
                    User Tasks
                  </li>
                </BreadcrumbItem>
              </ol>
            </nav>
          </Breadcrumb>
        </ComponentWithOuia>
      </Component>
    </section>
  </PageSection>
</UserTaskPageHeader>
`;
